ci/bin/format.dart

```diff
@@ -786,6 +798,133 @@ class GnFormatChecker extends FormatChecker {
   }
 }
 
+/// Checks the format of any .dart files using the "dart format" command.
+class DartFormatChecker extends FormatChecker {
+  DartFormatChecker({
+    super.processManager,
+    required super.baseGitRef,
+    required Directory repoDir,
+    super.allFiles,
+    super.messageCallback,
+  }) : super(
+    repoDir: repoDir,
+  ) {
+    // $ENGINE/flutter/third_party/dart/tools/sdks/dart-sdk/bin/dart
+    _dartBin = path.join(
+      repoDir.absolute.parent.path,
+      'flutter',
+      'third_party',
+      'dart',
+      'tools',
+      'sdks',
+      'dart-sdk',
+      'bin',
+      'dart',
+    );
+  }
+
+  late final String _dartBin;
+
+  @override
+  Future<bool> checkFormatting() async {
+    message('Checking Dart formatting...');
+    return (await _runDartFormat(fixing: false)) == 0;
+  }
+
+  @override
+  Future<bool> fixFormatting() async {
+    message('Fixing Dart formatting...');
+    await _runDartFormat(fixing: true);
+    // The dart formatter shouldn't fail when fixing errors.
+    return true;
+  }
+
+  Future<int> _runDartFormat({required bool fixing}) async {
+    final List<String> filesToCheck = await getFileList(<String>['*.dart']);
```

Comment:
This is a nit:

// Todo: we probably should just do this on single list of files rather than forking X processes.

- dart formatter is pretty quick 
- engineers probably aren't going to make huge changes
- most of the time is probably spent in process start

Reply:
I had this modeled after the other language formatters in this file, which all use this pattern. It has the advantage of allowing us to print a nice diff between expected formatting vs. actual formatting. It is probably faster to just pass the list of files directly to the formatter.

Reply:
Printing a nice diff seems like a good motivation to follow the same pattern.

Reply:
I don't think we _need_ to print a nice diff. We just _need_ to tell people the files that are failing format checking. They **should** be running et format before they put the PR up (which in the engine's case is true).